File: Java/CodingBat/maxEnd3.java
// http://codingbat.com/prob/p146256

public int[] maxEnd3(int[] nums) {
  int max = (nums[0] > nums[nums.length-1]) ? nums[0] : nums[nums.length-1];
  for (int i = 0; i < nums.length; ++i) {
    nums[i] = max;
  }
  return nums;
}

File: Python/CodingBat/max_end3.py
# http://codingbat.com/prob/p135290

def max_end3(nums):
  max = nums[0] if (nums[0] > nums[-1]) else nums[-1] # or use max(arg1, arg2)
  for i in range(3):
    nums[i] = max
  return nums
